src/components/pickers/Suggestions/Suggestions.tsx

```diff
@@ -80,37 +86,28 @@ export class Suggestions<T> extends BaseComponent<ISuggestionsProps<T>, {}> {
     }
   }
 
-  private _renderSuggestions(): JSX.Element[] {
+  private _renderSuggestions(): JSX.Element {
     let {
       suggestions,
       onRenderSuggestion,
-      noResultsFoundText,
       suggestionsItemClassName } = this.props;
-
-    if (!suggestions || !suggestions.length) {
-      return [<div className='ms-Suggestions-none'> { noResultsFoundText ? noResultsFoundText : ' ' } </div>];
-    }
-
-    let suggestionItems: JSX.Element[] = [];
     let TypedSuggestionsItem = this.SuggestionsItemOfProperType;
 
-    for (let index: number = 0; index <= suggestions.length - 1; index++) {
-      let suggestionItem: ISuggestionModel<T> = suggestions[index];
-      suggestionItems.push(
-        <div ref={ this._resolveRef(suggestionItem.isSelected ? '_selectedElement' : '') }
-          key={ index }
-          id={ 'sug-' + index }
-          role='menuitem'>
-          <TypedSuggestionsItem
-            suggestionModel={ suggestionItem }
-            RenderSuggestion={ onRenderSuggestion }
-            onClick={ (ev: React.MouseEvent) => this.props.onSuggestionClick(ev, suggestionItem.item, index) }
-            className={ suggestionsItemClassName }
-            />
-        </div>);
-    }
-
-    return suggestionItems;
+    return (
+      <div className='ms-Suggestions-container' id='suggestion-list' role='menu'>
+        { suggestions.map((suggestion, index) =>
+          <div ref={ this._resolveRef(suggestion.isSelected ? '_selectedElement' : '') }
+            key={ index }
```

Comment:
index is a bad key to use. If TypedSuggestionsItem has state, it will not reset if the array gets a new item inserted at index 0.

Is it possible that we could have suggestion.key or something unique to key off of?